boxoffice/models/item.py

```diff
@@ -43,6 +43,12 @@ def current_price(self):
         """
         return self.price_at(datetime.utcnow())
 
+    def has_higher_price(self, current_price):
```

Comment:
Not an immediate priority, but all these queries to render a widget are expensive. This value can be cached for up to a week (in our current use), so it needs to be pushed to cache for whatever timeline is left for validity.

I suggest model-level methods should return status and valid_until, and a wrapping function in a workflow object should push this to cache for the valid_until period.